### ISSUE TITLE
Do we need support for microphone or game controller sensor?

### ISSUE THREAD

**Issue by happyharryh** [moonlight-stream/moonlight-qt#883]
Due to the limitation of NVIDIA GameStream, there is neither microphone support nor game controller sensor support in moonlight-qt now. To support them, I have developed two branches, which embed a [VBan Emitter](https://github.com/happyharryh/moonlight-qt/tree/vban-emitter) and a [CemuHook Server](https://github.com/happyharryh/moonlight-qt/tree/cemuhook-server). They send voice data and motion data of game controllers (both accelerometer and gyro) to the host pc, so that I can enjoy the voice chat and motion sensing games remotely.
These two features give me a lot of fun indeed, such as playing Just Dance in Yuzu together with friends far away. However, I have not sent a pull request because I am afraid that they may be not suitable for the mainline version of moonlight-qt. The code writing maybe not comply with the rule of current framework of moonlight-qt. The compatibility and the stability have not been widely tested (I have only tested them on Windows platform using a Nintendo Pro controller and a pair of Joy-Cons). And what worries me the most, there may be few demands for these features. Perhaps, it is better to build them as plug-ins for those who need them?
Looking forward to your opinion.

**Comment by cgutman** [comment]
I like the plugin idea. There have been a few requests for features that would be nice to have while streaming but aren't really a good fit for Moonlight to implement itself.

**Comment by happyharryh** [comment]
> I like the plugin idea. There have been a few requests for features that would be nice to have while streaming but aren't really a good fit for Moonlight to implement itself.

Yeah, I like the idea, too. So, this Issue should become a feature request for the interface of plug-ins. Hope it will be realized someday. 😄
